pkg/server/getRecommendations.go

```diff
@@ -0,0 +1,137 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/golang/gddo/httputil/header"
+	"github.com/stolostron/recommends/pkg/config"
+	"github.com/stolostron/recommends/pkg/helpers"
+	"github.com/stolostron/recommends/pkg/utils"
+
+	klog "k8s.io/klog/v2"
+)
+
+var list_recommendations_url = config.Cfg.KruizeURL + "/listRecommendations"
+
+//struct to capture input for getRecommendations request
+type RecommendationInput []struct {
+	RecommendationId string `json:"recommendation_id"`
+	Namespace        string `json:"namespace"`
+	ClusterName      string `json:"cluster_name"`
+}
+
+func getRecommendations(w http.ResponseWriter, r *http.Request) {
+	klog.Infof("Received Request for list Recommendations")
+
+	var getRecommendations RecommendationInput
+	var requestUrlList []string
+	var recommendations []ListRecommendations
+	var RecommendationStatusMaps = RecommendationStatusMap{RecommendationStatus: make(map[string]string)}
+
+	//check content type is json
+	if r.Header.Get("Content-Type") != "" {
+		value, _ := header.ParseValueAndParams(r.Header, "Content-Type")
+		if value != "application/json" {
+			msg := "Content-Type header is not application/json"
+			http.Error(w, msg, http.StatusUnsupportedMediaType)
+			return
+		}
+	}
+
+	//decode user's input:
+	dec := json.NewDecoder(r.Body)
+	err := dec.Decode(&getRecommendations)
+
+	if ok := helpers.ErrorHandlingRequests(w, err); !ok {
+		return
+	}
+
+	recommendationId := getRecommendations[0].RecommendationId
+	namespace := getRecommendations[0].Namespace
+	clusterName := getRecommendations[0].ClusterName
+
+	//if recommendationid provided default to use recommendation id
+	if recommendationId != "" {
+		reqParts := strings.Split(recommendationId, "_")
+		id := reqParts[3]
+
+		for _, RecommendationIDMap := range RecommendationIDMaps {
+			rec := RecommendationIDMap.RecommendationID[id]
+			containerRequestUrl := fmt.Sprint(list_recommendations_url + "?" + "experiment_name=" + rec)
+			requestUrlList = append(requestUrlList, containerRequestUrl)
+		}
+	}
+	//if recommendationid is missing but clustername and namespace provided:
+	if namespace != "" && clusterName != "" && recommendationId == "" {
+		var id string
+		clusterNamespace := clusterName + "_" + namespace
+
+		id = ClusterNamespaceMaps[0].ClusterNamespace[clusterNamespace]
```

Comment:
ClusterNamespaceMaps[0] - Do we need a list of maps ? always getting the 1st element , seems can avoid the list @Anxhela21